`python39Packages.fakeredis` is currently broken

### Describe the bug
As of 4d922730369d1c468cc4ef5b2fc30fd4200530e0, the build of `python39Packages.fakeredis` is broken. This is currently breaking the build of other packages including 
* `python3Packages.wandb`
* `python3Packages.sentry-sdk`
* `cura`
* `moz-phab`
* `octoprint`
* `gns3-server`
* `gns3-gui`
* `tribler`
* `python3Packages.django-rq`
* `alerta-server`
* `moodle-dl`
and certainly many others...

~I suspect this was triggered by https://github.com/NixOS/nixpkgs/pull/140482, #131015, or one of the redis upgrades.~ I believe this was triggered by #152550.

### Steps To Reproduce
Steps to reproduce the behavior:
1. Checkout nixpkgs at 4d922730369d1c468cc4ef5b2fc30fd4200530e0
2. `nix-build -A python39Packages.fakeredis`
3. Observe failure
```
...
ERROR: Could not find a version that satisfies the requirement redis<4.1.0 (from fakeredis) (from versions: none)
ERROR: No matching distribution found for redis<4.1.0
```

### Expected behavior
The build to succeed.

### Screenshots
n/a

### Additional context
n/a

### Notify maintainers
<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->
@fabaff @SuperSandro2000 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.10.91, NixOS, 21.11 (Porcupine)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.16`
 - channels(root): `"nixos-21.11.335376.610d4ea2750, nix-ld"`
 - channels(skainswo): `"home-manager, nixpkgs-unstable-22.05pre343295.adf7f03d3bf"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

I tried patching around the issue with
```nix
  patchPhase = ''
    substituteInPlace setup.cfg --replace "redis<4.1.0" "redis"
  '';
```
but that gives me a bunch of test failures:
```
self = FakeStrictRedis<ConnectionPool<FakeConnection<server=<fakeredis._server.FakeServer object at 0x7ffff50060d0>,db=0>>>
connection = FakeConnection<server=<fakeredis._server.FakeServer object at 0x7ffff50060d0>,db=0>
command_name = 'DUMP', options = {'NEVER_DECODE': []}

    def parse_response(self, connection, command_name, **options):
        "Parses a response from the Redis server"
        try:
            if NEVER_DECODE in options:
>               response = connection.read_response(disable_decoding=True)
E               TypeError: read_response() got an unexpected keyword argument 'disable_decoding'

/nix/store/lz2jbzq4yaas58x3m8y83nxik37xp73z-python3.9-redis-4.1.0/lib/python3.9/site-packages/redis/client.py:1187: TypeError
```

--- a/pkgs/development/python-modules/chalice/default.nix
+++ b/pkgs/development/python-modules/chalice/default.nix
@@ -61,7 +61,8 @@ buildPythonPackage rec {
   postPatch = ''
     sed -i setup.py -e "/pip>=/c\'pip',"
     substituteInPlace setup.py \
-      --replace 'typing==3.6.4' 'typing'
+      --replace "typing==3.6.4" "typing" \
+      --replace "attrs>=19.3.0,<21.3.0" "attrs"
   '';
 
   disabledTestPaths = [

--- a/pkgs/development/python-modules/fakeredis/default.nix
+++ b/pkgs/development/python-modules/fakeredis/default.nix
@@ -2,7 +2,8 @@
 , aioredis
 , async_generator
 , buildPythonPackage
-, fetchPypi
+, fetchFromGitHub
+, fetchpatch
 , hypothesis
 , lupa
 , pytest-asyncio
@@ -19,11 +20,13 @@ buildPythonPackage rec {
   version = "1.7.0";
   format = "setuptools";
 
-  disabled = pythonOlder "3.5";
+  disabled = pythonOlder "3.7";
 
-  src = fetchPypi {
-    inherit pname version;
-    sha256 = "sha256-yb0S5DAzbL0+GJ+uDpHrmZl7k+dtv91u1n+jUtxoTHE=";
+  src = fetchFromGitHub {
+    owner = "jamesls";
+    repo = pname;
+    rev = version;
+    hash = "sha256-P6PUg9SY0Qshlvj+iV1xdrzVLJ9JXUV4cGHUynKO3m0=";
   };
 
   propagatedBuildInputs = [
@@ -40,6 +43,15 @@ buildPythonPackage rec {
     pytest-asyncio
     pytest-mock
     pytestCheckHook
+  ];
+
+  patches = [
+    # Support for redis >= 4.1.0, https://github.com/jamesls/fakeredis/pull/324
+    (fetchpatch {
+      name = "support-redis-4.1.0.patch";
+      url = "https://github.com/jamesls/fakeredis/commit/8ef8dc6dacc9baf571d66a25ffbf0fadd7c70f78.patch";
+      sha256 = "sha256-4DrF/5WEWQWlJZtAi4qobMDyRAAcO/weHIaK9waN00k=";
+    })
   ];
 
   disabledTestPaths = [

--- a/pkgs/development/python-modules/fakeredis/default.nix
+++ b/pkgs/development/python-modules/fakeredis/default.nix
@@ -46,7 +46,7 @@ buildPythonPackage rec {
   ];
 
   patches = [
-    # Support for redis >= 4.1.0, https://github.com/jamesls/fakeredis/pull/324
+    # Support for redis <= 4.1.0, https://github.com/jamesls/fakeredis/pull/324
     (fetchpatch {
       name = "support-redis-4.1.0.patch";
       url = "https://github.com/jamesls/fakeredis/commit/8ef8dc6dacc9baf571d66a25ffbf0fadd7c70f78.patch";

